### PR TITLE
change flag to mutex lock for native-audio

### PR DESCRIPTION
**Pull request by ggfan**

### CODE REVIEW

File: native-audio/app/src/main/jni/native-audio-jni.c [hunk 1]
```diff
@@ -668,7 +675,7 @@ jboolean Java_com_example_nativeaudio_NativeAudio_selectClip(JNIEnv* env, jclass
         SLresult result;
         result = (*bqPlayerBufferQueue)->Enqueue(bqPlayerBufferQueue, nextBuffer, nextSize);
         if (SL_RESULT_SUCCESS != result) {
-            bqPlayerRecorderBusy = 1;
+            pthread_mutex_unlock(&audioEngineLock);
             return JNI_FALSE;
         }
```
Comment (hak) [inline native-audio/app/src/main/jni/native-audio-jni.c]: If Enqueue fails, is it still locked?

Comment (ggfan) [inline native-audio/app/src/main/jni/native-audio-jni.c]: the above code is unlock, so it is free for re-entry

Comment (hak) [inline native-audio/app/src/main/jni/native-audio-jni.c]: Had offline discussion and now I get the idea behind the change.

Just one point,
'bqPlayerRecorderBusy' was more descriptive than 'audioEngineLock' what's it's doing.
Can we do either:
- change 'audioEngineLock' name something like 'mutex_recorder_busy'?
or
- add some description what the mutex is for?

**Comment by hak** [comment]
LGTM

**Pull request merged**